client/src/sections/resources/Melodramedary.tsx

```diff
@@ -0,0 +1,218 @@
+    useGet("questL10Garbage") in ["step10", "finished"];
+  const gunpowderRequirements =
+    useGet("sidequestLighthouseCompleted") === "none" && gsbRequirements;
+  const evilEyeRequirements = !(useGet("cyrptNookEvilness") < 35);
+  const mojoFilterRequirements = useGet("desertExploration") > 0;
+
+  // This long statement attempts to build out recommendations. Here are the rankings I used.
+  //   - #1: Bowling Balls (4 turns, if you need 4 balls)
+  //   - #2: GSBs (3.6 turns)
+  //   - #3: Star Key (~3 turns, hard to value!)
+  //   - #4: Barrels of Gunpowder (2 turns, saves 2 backups as a nostalgia-spit)
+  //   - #5: Evil Eyes (2-ish turns, hard to value!)
+  //   - #6: Mojo Filters (Very good in unrestricted paths in CMC meta, likely 2.5-3 turns but 1 in standard, if even that)
+
+  const allSpitTargets = [
+    new SpitTarget(
+      "Bowling Balls",
+      "Pygmy Bowlers",
+      "The Hidden Bowling Alley",
+      11,
+      bowlingRequirements
+    ),
+    new SpitTarget(
+      "Green Smoke Bombs",
+      "Green Ops Soldier",
+      "The Battlefield",
+      12,
+      gsbRequirements
+    ),
+    new SpitTarget(
+      "8 stars + 8 lines",
+      hitsMonster,
+      "The Hole in the Sky",
+      10,
+      starKeyRequirements
+    ),
+    new SpitTarget(
+      "Barrels of Gunpowder",
+      "Lobsterfrogman",
+      "Sonofa Beach",
+      12,
+      gunpowderRequirements,
+      2
+    ),
+    new SpitTarget(
+      "Evil Eyes",
+      "All non-partying Skeletons",
+      "The Defiled Nook",
+      7,
+      evilEyeRequirements,
+      2
+    ),
+    new SpitTarget(
+      "Mojo Filters",
+      "A Swarm of Scarab Beatles",
+      "The Oasis",
+      11,
+      mojoFilterRequirements
+    ),
+  ];
+
+  // We will only display the top 3 recommendations; iterate through the list and stop when recs are full
+
+  let recommendations: SpitTarget[] = [];
+  for (const target of allSpitTargets) {
+    if (recommendations.length === 3) {
+      break; // This only populates 3. There's probably a better way to do this?
+    }
+    if (target.accessible(userLevel)) {
+      recommendations.push(target);
+    }
+  }
+
+  // My to-do list on this tile:
+  //   -- Figure out what to suggest if no spits accessible
+  //   -- Figure out how to add tooltips; then, pass both .formatString and .toolTip via spitTarget
+
+  return (
+    <Tile
+      header="Melodramedary"
+      imageUrl="/images/otherimages/Camelfam_left.gif"
+      hide={!useHave($familiar`Melodramedary`)}
+    >
+      {spitProgress < 100 && (
+        <Line>Current spit progress: {spitProgress}%</Line>
+      )}
+      {spitProgress < 100 && haveDrinkingHelmet && (
+        <Line>
+          {floor((100 - spitProgress) / 4.5, 0)} combats until your next spit.
+        </Line>
+      )}
+      {spitProgress < 100 && haveDrinkingHelmet && (
+        <Line>
+          {floor((100 - spitProgress) / 3, 0)} combats until your next spit.
+        </Line>
+      )}
```

Comment:
here you could just move the haveDrinkingHelmet conditional inside the expression, it would probably be better style even to move the computation above the return statement.
```suggestion
      {spitProgress < 100 && (
        <Line>
          {floor((100 - spitProgress) / (haveDrinkingHelmet ? 4.5 : 3), 0)} combats until your next spit.
        </Line>
      )}
```